### PR TITLE
Add externalId

### PR DESCRIPTION
**Pull request by satiani**

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collectionsheet/data/JLGCollectionSheetFlatData.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public BigDecimal getChargesDue() {
     }
 
     public LoanDueData getLoanDueData() {
-        return new LoanDueData(this.loanId, this.accountId, this.accountStatusId, this.productShortName, this.productId, this.currency,
+        return new LoanDueData(this.loanId, null, this.accountId, this.accountStatusId, this.productShortName, this.productId, this.currency,
```
Comment (blmarland) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collectionsheet/data/JLGCollectionSheetFlatData.java]: why  null here?

Comment (satiani) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/collectionsheet/data/JLGCollectionSheetFlatData.java]: Due to utter laziness :) We're not using this product in Mifos and adding externalId here will involve large changes

**Comment by blmarland** [comment]
where is external id actually linked to campaign ID?

**Comment by satiani** [comment]
During import stage they are appended to clients.

**Comment by blmarland** [comment]
I dont think I can give an LGTM on this, its a ton of code that i know nothing about in languages I haven't seen before.

**Pull request merged**